snuba/query/validation/validators.py

```diff
@@ -52,3 +55,69 @@ def validate(self, query: Query, alias: Optional[str] = None) -> None:
             raise InvalidQueryException(
                 f"missing required conditions for {', '.join(missing)}"
             )
+
+
+class OneProjectValidator(QueryValidator):
+    def validate(self, query: Query, alias: Optional[str] = None) -> None:
+        match = build_match("project_id", list(FUNCTION_TO_OPERATOR.keys()), int)
+        condition = query.get_condition()
+        top_level = get_first_level_and_conditions(condition) if condition else []
+        matched_id: Optional[int] = None
+        for cond in top_level:
+            result = match.match(cond)
+            if result is not None:
+                rhs = result.expression("rhs")
+                project_id: Optional[int] = None
+                if isinstance(rhs, Literal) and isinstance(rhs.value, int):
+                    project_id = rhs.value
+                elif isinstance(rhs, FunctionCall):
+                    project_ids = set(rhs.parameters)
+                    if len(project_ids) != 1:
+                        raise InvalidQueryException("Must only query one project")
+                    value = project_ids.pop()
+                    assert isinstance(value, Literal) and isinstance(value.value, int)
+                    project_id = value.value
```

Comment:
You should probably raise InvalidQueryException if the set does not contain a Literal that contains an integer.